src/Aspire.Dashboard/Components/ResourcesGridColumns/GridColumnManager.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Components.Resize;
+using Aspire.Dashboard.Model;
+
+namespace Aspire.Dashboard.Components;
+
+public class GridColumnManager : IDisposable
+{
+    private readonly GridColumn[] _columns;
+    private readonly DimensionManager _dimensionManager;
+    private ViewportInformation _currentViewport;
+
+    public GridColumnManager(GridColumn[] columns, ViewportInformation viewportInformation, DimensionManager dimensionManager)
```

Comment:
Nit: ViewportInformation should be available as a property on DimensionManager. If that is done then it doesn't need to be explicitly set in each manager's constructor.

Reply:
changed